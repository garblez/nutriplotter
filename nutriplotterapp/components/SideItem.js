import React from 'react';
import { Text, ImageBackground, StyleSheet, TouchableOpacity } from 'react-native';

export default class SideItem extends React.Component {

state = {
	x: 0,
	y: 0,
}
constructor(props){
	super(props);
}

componentDidMount(){
	this.setState({x: this.props.x, y: this.props.y})
}
render() {
    return(
		<TouchableOpacity>
			<ImageBackground
				alignContent={'center'}
				style={styles.image}
				source={require('../assets/images/apple.png')}>
				<Text style={styles.text}>+</Text>
			</ImageBackground>
		</TouchableOpacity>
	)
  }
}

const styles = StyleSheet.create ({
	image: {
		flex: 1,
		width: 64,
		height: 64,
		margin: 5,
		zIndex: 10,
  },
	text: {
		flex: 1,
		textAlign: "center",
		justifyContent: "center",
		alignItems: "center",
		zIndex: 11,
	},
	modal: {
		flex: 1,
	},
});