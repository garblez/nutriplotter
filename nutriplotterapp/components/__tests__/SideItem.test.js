import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import SideItem from '../SideItem.js';
import PopulateSideItems from '../../PopulateSideItems.js';
var r = new PopulateSideItems();

//first test to ensure working correctly
it('Test conditions working properly', async () => {
    const testVar = true;
    expect(true).toEqual(testVar);
});


it('SideItem: renders correctly', () => {
  const tree = renderer.create(<SideItem type={'fruit'} isDown={false} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('SideItem: should set passed values to state', () => {
  let sideItemComponent = renderer.create(<SideItem type={'fruit'} isDown={false} updateStateHome={0} />).getInstance()
  const targetState = {
  	isDown: false,
  	isPressed: false,
  	type: 'fruit',
  	refresh: 0,
  };
  expect(sideItemComponent.state).toEqual(targetState);
});

it('SideItem: isPressed function', () => {
  let sideItemComponent = renderer.create(<SideItem type={'fruit'} isDown={false} />).getInstance()
  sideItemComponent.sideItemPressed();
  const targetState = {
  	isDown: false,
  	isPressed: true,
  	type: 'fruit',
  	refresh: 0,
  };
  expect(sideItemComponent.state).toEqual(targetState);
});
