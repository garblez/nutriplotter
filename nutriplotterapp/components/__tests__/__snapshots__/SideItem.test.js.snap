// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideItem: renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "height": 64,
          "justifyContent": "center",
          "margin": 5,
          "textAlign": "center",
          "width": 64,
          "zIndex": 10,
        }
      }
    >
      <Image
        source={1}
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": 64,
              "width": 64,
            },
            undefined,
          ]
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "textAlign": "center",
            "zIndex": 11,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 24,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
